src/dynamics/video_player/player/player.js

```diff
@@ -1286,45 +1294,58 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
                         ctx.clearRect(0, 0, canvas.width, canvas.height)
                         ctx.drawImage(video, 0, 0, canvas.width, canvas.height);
                         cb(canvas, ctx);
-                    }.bind(this), 200);
+                    }.bind(this), 250);
+                },
+                removeImgElements: function(imgElements) {
+                    imgElements.forEach(img => {
+                        if (img.parentNode)
+                            img.parentNode.removeChild(img);
+                    })
                 },
-                _renderVideoFrame: function(video) {
+                addSourcToVideo: function(source, video) {
                     const currentPosition = this.getCurrentPosition();
-                    video.style.backgroundColor = 'transparent';
+                    this.vidEle.src = source;
                     const videoParentEle = video.parentElement;
+                    this.imgEle.style.width = '100%';
+                    this.vidEle.setAttribute('crossorigin', 'anonymous')
+                    this.vidEle.muted = true;
+                    this.set('trackFrameTime', 0);
 
-                    img = document.createElement('img');
-                    img.style.width = '100%';
-                    const imgElements = videoParentEle.querySelectorAll('img');
-
-                    const vidEle = document.createElement('video');
-                    vidEle.src = this.get("source");
-                    vidEle.setAttribute('crossorigin', 'anonymous')
-                    vidEle.muted = true;
-                    vidEle.play();
-
-
+                    this.vidEle.addEventListener("loadeddata", (event) => {
+                        this.vidEle.pause();
+                        this._drawFrame(this.vidEle, currentPosition, video.videoWidth, video.videoHeight, (canvas) => {
+                            this.removeImgElements(videoParentEle.querySelectorAll('img'));
+                            this.imgEle.src = `${canvas.toDataURL()}`;
+                            videoParentEle.appendChild(this.imgEle);
```

Comment:
Hi @maxblessed, do we really need to remove and append the img element again? Changing the src value should be enough
```suggestion
                            if (this.imgEle) {
                                this.imgEle.src = `${canvas.toDataURL()}`;
                            } else {
                                this.imgEle = document.createElement('img');
                                this.imgEle.style.width = '100%';
                                this.imgEle.src = `${canvas.toDataURL()}`;
                                videoParentEle.appendChild(this.imgEle);
                            }
```